src/commtrack.js

```diff
@@ -0,0 +1,377 @@
+                    "section-id": mug.p.sectionId,
+                    "vellum:role": "Transfer"
+                };
+            },
+            icon: 'icon-exchange',
+            init: function (mug, form) {
+                mug.p.src = {value: ""};
+                mug.p.dest = {value: ""};
+                mug.p.sectionId = "";
+                mug.p.entryId = {value: ""};
+                mug.p.quantity = "";
+                mug.p.date = {value: "today()"};
+            },
+            spec: {
+                xmlnsAttr: { presence: "optional" },
+                src: {
+                    lstring: 'Source Case',
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: setValueWidget,
+                    xpathType: "generic",
+                    help: 'XPath expression for the case ID issuing the transaction. Leave blank if unknown or not applicable.',
+                    validationFunc: transferMugValidation,
+                },
+                dest: {
+                    lstring: 'Destination Case',
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: setValueWidget,
+                    xpathType: "generic",
+                    help: 'XPath expression for the case ID receiving the transaction. Leave blank if unknown or not applicable.',
+                    validationFunc: transferMugValidation,
+                },
+                sectionId: {
+                    lstring: 'Balance ID',
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: widgets.text,
+                    help: 'The name of the balance you are tracking. ' + 
+                         'This is an internal identifier which does not appear on the phone.',
+                },
+                entryId: {
+                    lstring: 'Product',
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: setValueWidget,
+                    xpathType: "generic",
+                    help: 'A reference to a product ID, e.g., "/data/products/current_product"',
+                },
+                quantity: {
+                    lstring: 'Quantity',
+                    visibility: 'visible',
+                    presence: 'optional',
+                    widget: widgets.xPath,
+                    xpathType: "generic",
+                    help: 'A reference to an integer question in this form.',
+                },
+                requiredAttr: { presence: "notallowed" },
+                constraintAttr: { presence : "notallowed" },
+                calculateAttr: { presence: "notallowed" }
+            }
+        };
+
+    $.vellum.plugin("commtrack", {}, {
+        getAdvancedQuestions: function () {
+            var questions = this.__callOld();
+            if (this.opts().features.transaction_question_types) {
```

Comment:
I feel like this is probably not the best place to check the feature flag - like it should be somewhere much higher up - but wasn't successful. Open to suggestions.